src/lib.rs

```diff
@@ -402,30 +544,23 @@ fn parse_outpoint(s: &str) -> Result<OutPoint, String> {
     OutPoint::from_str(s).map_err(|e| format!("{:?}", e))
 }
 
-/// Execute a wallet sub-command with a given [`Wallet`].
+/// Execute an offline wallet sub-command
 ///
-/// Wallet sub-commands are described in [`WalletSubCommand`]. See [`crate`] for example usage.
+/// Offline wallet sub-commands are described in [`OfflineWalletSubCommand`].
 #[maybe_async]
-pub fn handle_wallet_subcommand<C, D>(
-    wallet: &Wallet<C, D>,
-    wallet_subcommand: WalletSubCommand,
+pub fn handle_offline_wallet_subcommand<D>(
+    wallet: &Wallet<(), D>,
```

Comment:
You could change this to take a generic type `T` instead of just `()`, which would allow you to call this with a ref to an online wallet

```suggestion
pub fn handle_offline_wallet_subcommand<T, D>(
    wallet: &Wallet<T, D>,
```